src/caffeine/caffeine_assert_s.F90

```diff
@@ -0,0 +1,36 @@
+! Copyright (c), The Regents of the University of California
+! Terms of use are as specified in LICENSE.txt
+submodule(prif:prif_private_s) caffeine_assert_s
+  implicit none
+
+#if CAF_ASSERTIONS || !defined(CAF_ASSERTIONS)
+  logical, parameter :: assertions_=.true.
+#else
+  logical, parameter :: assertions_=.false.
+#endif
+
+   !! Disable assertions with
+   !! fpm test --flag "-DCAF_ASSERTIONS=0"
```

Comment:
The comment above seems to imply that's a working command, but in my testing that's a non-functional command in our repo, for reasons entirely unrelated to this module. Something like `./build/run-fpm.sh test --flag "-DCAF_ASSERTIONS=0"` would be closer, but even that doesn't actually work due to the semantics of `fpm --flag`. 

There may not currently be a concise build command that actually toggles these assertions, but even if there were that probably belongs in documentation like README, not as a comment in the assert module. I suggest we replace this comment with something more generic like:

```suggestion
   !! Disable assertions by compiling with preprocessor setting: -DCAF_ASSERTIONS=0
```

CC: @everythingfunctional